cli/npm/resolution.rs

```diff
@@ -415,32 +424,49 @@ impl NpmResolutionSnapshot {
           dependencies,
         };
 
-        packages.insert(package_id.clone(), package);
+        packages.insert(package_id, package);
+      }
+
+      // verify that all these ids exist in packages
+      for id in &verify_ids {
+        if !packages.contains_key(id) {
+          bail!(
+            "the lockfile ({}) is corrupt. Recreate it or run again with --no-lock",
```

Comment:
We don't have `--no-lock` flag. Did you mean `You can recreate it with `--lock-write` flag